### PR TITLE
chore: merge google-cloud-java's java-shared-dependencies and java-core

### PR DESCRIPTION
**Pull request by suztomo**
Merging google-cloud-java's java-shared-dependencies and java-core modules into gapic-generator-java repository.

Note for myself: when merging, **DO NOT SQUASH**.

TODO: after this pull request: move relevant checks from google-cloud-java to gapic-generator-java.

TODO: 1 hour after merge, confirm the release pull request has correct version bumps for the two modules.

Not in scope: I don't intend to fix "[gapic-generator-java-root] SonarCloud Code Analysis" at least in this pull request. -> https://github.com/googleapis/gapic-generator-java/issues/1521

# How I created this pull request (to preserve commit history)

Setup for `git filter-repo`

```
git clone https://github.com/newren/git-filter-repo.git
export PATH=$PATH:`pwd`/git-filter-repo
```

For `service` in java-shared-dependencies and java-core, I ran the following commands:

```
cd ~
git clone https://github.com/googleapis/google-cloud-java google-cloud-java-$service
cd ~/google-cloud-java-$service
git filter-repo --subdirectory-filter $service --to-subdirectory-filter $service
cd ~/gapic-generator-java
git remote add ${service} "$HOME/google-cloud-java-$service"
git config --add secrets.allowed "dest.*src"
git fetch ${service}
EDITOR=true git merge --quiet --allow-unrelated-histories ${service}/main
```

Reference:
- https://github.com/googleapis/google-cloud-java/blob/main/generation/merge_repository.sh
- https://htmlpreview.github.io/?https://github.com/newren/git-filter-repo/blob/docs/html/git-filter-repo.html

### CODE REVIEW

**Comment by suztomo** [comment]
I'll temporarily cla check that is failing due to too many commits.

File: versions.txt [hunk 1]
```diff
@@ -15,3 +15,5 @@ grpc-google-iam-v2beta:1.9.3:1.9.4-SNAPSHOT
 google-iam-policy:1.9.3:1.9.4-SNAPSHOT
 proto-google-iam-v2:1.9.3:1.9.4-SNAPSHOT
 grpc-google-iam-v2:1.9.3:1.9.4-SNAPSHOT
+google-cloud-core:2.13.0:2.14.0-SNAPSHOT
+google-cloud-shared-dependencies:3.5.0:3.6.0-SNAPSHOT
```
Comment (suztomo) [inline versions.txt:19]: They are from https://github.com/googleapis/google-cloud-java/blob/e7d648dd992a59df16cbdeb47623511f5ebb6d19/versions.txt#L606

- The latest google-cloud-core is 2.13.0 https://central.sonatype.com/artifact/com.google.cloud/google-cloud-core/2.13.0
- The latest shared dependencies BOM is 3.5.0 https://central.sonatype.com/artifact/com.google.cloud/google-cloud-shared-dependencies/3.5.0

File: java-shared-dependencies/dependency-convergence-check/pom.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
+  </properties>
+
+  <dependencyManagement>
+    <dependencies>
+    </dependencies>
+  </dependencyManagement>
```
[the remark this answers is not in the document]
Comment (suztomo) [reply java-shared-dependencies/dependency-convergence-check/pom.xml]: Removed.

**Review by suztomo — COMMENTED** [review]
@mpeddada1 PTAL

**Review by mpeddada1 — APPROVED** [review]
LGTM

**Comment by suztomo** [comment]
Merged. The history is preserved. The cla check is back.

Screenshot 2023-03-22 at 1 33 39 PM

**Comment by suztomo** [comment]
> TODO: 1 hour after merge, confirm the release pull request has correct version bumps for the two modules.

https://github.com/googleapis/gapic-generator-java/pull/1511/files is showing:

For java-core:

Screenshot 2023-03-22 at 1 40 55 PM


For java-shared-dependencies:
Screenshot 2023-03-22 at 1 42 11 PM

**Pull request merged**